test/cases/neuron-inference/main_test.go

```
@@ -115,3 +117,41 @@ func discoverNeuronCoreCapacity(ctx context.Context, config *envconf.Config) (co
 	log.Println("[INFO] Neuron capacity discovery complete.")
 	return ctx, nil
 }
+
+func getNodeCapacity(ctx context.Context, config *envconf.Config) (context.Context, error) {
+	cs, err := kubernetes.NewForConfig(config.Client().RESTConfig())
+	if err != nil {
+		return ctx, fmt.Errorf("failed to create kubernetes client: %w", err)
+	}
+
+	nodes, err := cs.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return ctx, fmt.Errorf("failed to list nodes: %w", err)
+	}
+	if len(nodes.Items) == 0 {
+		return ctx, fmt.Errorf("no nodes found in the cluster")
+	}
+	var totalNeuron, totalNeuronCore int
+	for _, node := range nodes.Items {
+		instanceType := node.Labels["node.kubernetes.io/instance-type"]
+		neuronCap, hasNeuron := node.Status.Capacity["aws.amazon.com/neuron"]
+		neuronCoreCap, hasNeuronCore := node.Status.Capacity["aws.amazon.com/neuroncore"]
+		if hasNeuron {
+			totalNeuron += int(neuronCap.Value())
+		} else {
+			log.Printf("[WARN] Node %s (type=%s) lacks 'aws.amazon.com/neuron'.", node.Name, instanceType)
+		}
+		if hasNeuronCore {
+			totalNeuronCore += int(neuronCoreCap.Value())
+		} else {
+			log.Printf("[WARN] Node %s (type=%s) lacks 'aws.amazon.com/neuroncore'.", node.Name, instanceType)
+		}
+	}
+	nodeCount := len(nodes.Items)
+	if nodeCount > 0 {
+		neuronPerNode = totalNeuron / nodeCount
+		neuronCorePerNode = totalNeuronCore / nodeCount
+	}
```

we should probably just return an error here if nodeCount in the else condition

---

Revised